### DISCUSSION TITLE
How to get root element from very deep child DisplayObject?

### DISCUSSION THREAD

**Discussion by humbleCoder1990** [pixijs/pixijs#7710] [Q&A]
I need to know root element (PIXI.Stage)'s width & height from InteractionEvent.target (DisplayObject).
I read https://www.html5gamedevs.com/topic/35138-access-renderer-from-child-classes/
Now it's v6. Is there any way to get root element?

**Comment by ivanpopelyshev** [comment]
`renderer.screen`  or some other object that you globally manage . There's no other way. If you cant do that - well, you have an architectural problem with your app. you know that `stage.width` will actually calculate bounds , right?

**Comment by miltoncandelero** [comment]
`renderer.view` or if using `Application` it might be something like `app.renderer.view` or even `app.view`.

That will return the DOM Canvas Element. If you just want to know the `width` and `height` of that canvas element then `renderer.width` and `renderer.height` are exactly that.
